src/redux/slices/userSlice.js

```diff
@@ -1,20 +1,288 @@
 import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
 import axios from 'axios';
 
+import checkToken from '../../utilities/checkToken';
+
+const API_URL = import.meta.env.VITE_API_URL || 'http://localhost:3000';
+
+export const fetchUser = createAsyncThunk(
+  'user/fetchUser',
+  async (placeholder, { rejectWithValue }) => {
+    try {
+      const { token, user } = await checkToken();
+
+      const res = await axios.get(API_URL + `/api/user/${user.id}`, {
```

Comment:
Since you're pulling the user.id from the token, would this conflict with the params of the URL of the page the user is on? (This is badly worded I can explain more)

Reply:
yep good call - thought it'd be easier, but you're right -- need to remain flexible so we can pull info from other users